### PR TITLE
Cherrypick #7199 and mark MSBuildForUnity as WSA-only

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

Cherry-picks #7199 and additionally gates the MSBuildForUnity option behind the UWP build target.

image

## Changes
- Helps with #6947 by disabling MSBuild for Unity import unless the UWP build target is selected.

### CODE REVIEW

File: Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs [hunk 1]
```diff
@@ -52,46 +52,91 @@ public enum Configurations
...
-            { Configurations.ForceTextSerialization,  () => { return IsForceTextSerialization(); } },
-            { Configurations.VisibleMetaFiles,  () => { return IsVisibleMetaFiles(); } },
+            { Configurations.LatestScriptingRuntime, new ConfigGetter(() => { return IsLatestScriptingRuntime(); }) },
+            { Configurations.ForceTextSerialization, new ConfigGetter(() => { return IsForceTextSerialization(); }) },
+            { Configurations.VisibleMetaFiles, new ConfigGetter(() => { return IsVisibleMetaFiles(); }) },
             // Ensure compatibility with the pre-2019.3 XR architecture for customers / platforms
             // with legacy requirements.
 #pragma warning disable 0618
-            { Configurations.VirtualRealitySupported,  () => { return PlayerSettings.virtualRealitySupported; } },
+            { Configurations.VirtualRealitySupported, new ConfigGetter(() => { return PlayerSettings.virtualRealitySupported; }) },
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs]: can you update this to use the new XRSettingsUtilities.LegacyXRSupported property?

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs]: that will allow removing the pragma

Comment (keveleigh) [reply Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs]: ah yep yep

Comment (keveleigh) [reply Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs]: Updated

**Pull request merged**